realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/util/KotlinTestUtils.kt

```diff
@@ -0,0 +1,34 @@
+package io.realm.util
+
+import io.realm.ErrorCode
+import io.realm.ObjectServerError
+import org.junit.Assert.assertEquals
+import org.junit.Assert.fail
+
+// Helper methods for improving Kotlin unit tests.
+
+/**
+ * Verify that a specific exception is thrown
+ */
+inline fun <reified T : Exception> expectException(method: () -> Unit) {
```

Comment:
As already mentioned `org.jetbrains.kotlin:kotlin-test` has an `assertFailsWith` which we could use in Kotlin-land.